LinkedList.js

```diff
@@ -0,0 +1,174 @@
+import { LinkedListElement } from "./LinkedListElement.js";
+
+export class LinkedList {
+  #head = null;
+  #tail = null;
+
+  addToTail(value) {
+    let newMember = null;
+    console.log("adding " + value);
+    if (this.#head === null) {
+      this.#head = new LinkedListElement(value, null, null);
+      this.#tail = this.#head;
+    } else if (this.#head.nextMember === null) {
+      newMember = new LinkedListElement(value, null, this.#head);
+      this.#head.nextMember = newMember;
+    } else {
+      newMember = new LinkedListElement(value, null, this.#tail);
+      this.#tail.nextMember = newMember;
+    }
+    this.#tail = newMember;
+  }
+
+  toArray() {
+    let currentMember = this.#head;
+    let array = [];
+    while (currentMember !== null) {
+      array.push(currentMember);
+      currentMember = currentMember.nextMember;
+    }
+    return array;
+  }
+
+  remove(elementNumber) {
+    if (!isNaN(elementNumber) && elementNumber <= this.getLengthOfList()) {
+      const elementToRemove = this.getNthElement(elementNumber);
+      if (elementToRemove.nextMember) {
+        elementToRemove.nextMember.previousMember =
+          elementToRemove.previousMember;
+      } else {
+        this.#tail = elementToRemove.previousMember;
+      }
+      if (elementToRemove.previousMember) {
+        elementToRemove.previousMember.nextMember = elementToRemove.nextMember;
+      } else {
+        this.#head = elementToRemove.nextMember;
+      }
+    }
+  }
+
+  reverse = () => {
+    if (this.#tail && this.#head !== this.#tail) {
+      let previous = null;
+      const originalHead = this.#head;
+      while (this.#head !== null) {
+        let next = this.#head.nextMember;
+        this.#head.nextMember = previous;
+        this.#head.previousMember = next;
+        previous = this.#head;
+        this.#head = next;
+      }
+      this.#tail = originalHead;
+      this.#head = previous;
+    }
+  };
+  switchPositions(leftElement, rightElement) {
```

Comment:
please split this function into smaller ones, remember about good function names.

Please recall how we were working on the virtual world, the second time you implemented it was much better because you kept small functions and you had clean code in mind while coding. Do the same here.

Reply:
also a `switch case` might be a good solution here

Reply:
done